src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -477,12 +477,64 @@ private void processMultipleLinesInside(DetailAST ast) {
         if (isClassMemberBlock(astType)) {
             final List<Integer> emptyLines = getEmptyLines(ast);
             final List<Integer> emptyLinesToLog = getEmptyLinesToLog(emptyLines);
-
             for (Integer lineNo : emptyLinesToLog) {
-                // Checkstyle counts line numbers from 0 but IDE from 1
-                log(lineNo + 1, MSG_MULTIPLE_LINES_INSIDE);
+                log(getLastElementBeforeEmptyLines(ast, lineNo), MSG_MULTIPLE_LINES_INSIDE);
+            }
+        }
+    }
+
+    /**
+     * Returns the element after which empty lines exist.
+     *
+     * @param ast the ast to check.
+     * @param line the empty line which gives violation.
+     * @return The DetailAST after which empty lines are present.
+     */
+    private static DetailAST getLastElementBeforeEmptyLines(DetailAST ast, int line) {
+        DetailAST travelAST = ast;
+        if (ast.getFirstChild().getLineNo() <= line) {
+            travelAST = ast.getFirstChild();
+            while (travelAST.getNextSibling() != null
+                    && travelAST.getNextSibling().getLineNo() <= line) {
+                travelAST = travelAST.getNextSibling();
+            }
+            if (travelAST.hasChildren()) {
+                travelAST = getLastElementBeforeEmptyLines(travelAST, line);
```

Comment:
I see that there can be an issue when checking code like 
```
class Outer {
@Deprecated
@Override
class SomeClass {}




}
```

Because annotations are children of class_def.
Please extend test inputs and fix code if necessary

Reply:
Sir, I have fixed the issue, but wouldn't it be better to solve this in a new issue as this is completely unrelated to this issue?

Reply:
I think it is fine to fix it here